vowpalwabbit/example.h

```diff
@@ -133,7 +119,7 @@ struct vw;
 struct flat_example
 {
   polylabel l;
-  float weight = 1.f;  // a relative importance weight for the example, default = 1
```

Comment:
Why is weight needed in example, but not in flat_example?

Reply:
I am simply migrating back to old usage here. It's potentially a shortcoming in the flat_example type